ENIGMAsystem/SHELL/Graphics_Systems/General/GSscreen.cpp

```diff
@@ -432,57 +461,28 @@ int screen_save_part(string filename,unsigned x,unsigned y,unsigned w,unsigned h
   return ret;
 }
 
-int background_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload)
-{
-  draw_batch_flush(batch_flush_deferred);
-
-  bool flipped = false;
-  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
-
-  if (flipped)
-    rgba = enigma::image_flip(rgba, w, h, 4);
-
-  enigma::backgroundstructarray_reallocate();
-  int bckid=enigma::background_idmax;
-  enigma::background_new(bckid, w, h, &rgba[0], removeback, smooth, preload, false, 0, 0, 0, 0, 0, 0);
-  delete[] rgba;
-  enigma::background_idmax++;
-  return bckid;
+int background_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload) {
+  RawImage img = create_from_screen_helper(x, y, w, h, removeback, true);
+  Background bkg(w, h, graphics_create_texture(w, h, img.w, img.h, img.pxdata, false));
+  return backgrounds.add(std::move(bkg));
 }
 
 int sprite_create_from_screen(int x, int y, int w, int h, bool removeback, bool smooth, bool preload, int xorig, int yorig) {
-  draw_batch_flush(batch_flush_deferred);
-
-  bool flipped = false;
-  unsigned char* rgba = enigma::graphics_copy_screen_pixels(x,y,w,h,&flipped);
-
-  if (flipped)
-    rgba = enigma::image_flip(rgba, w, h, 4);
-
+  RawImage img = create_from_screen_helper(x, y, w, h, removeback, false);
```

Comment:
Why would background from screen do padding but not a sprite created from the screen?

Reply:
sprites have this code in add subimage